src/devices/Mcp3428/Mcp342x.cs

```diff
@@ -0,0 +1,387 @@
+        {
+            if (channel < 0 || channel > ChannelCount - 1)
+                throw new ArgumentOutOfRangeException(nameof(channel));
+            byte conf = 0;
+            var ok = true;
+            conf = Helpers.SetModeBit(conf, mode);
+            conf = Helpers.SetChannelBits(conf, channel);
+            conf = Helpers.SetGainBits(conf, pgaGain);
+            conf = Helpers.SetResolutionBits(conf, resolution);
+            conf = Helpers.SetReadyBit(conf, false);
+
+            _i2cDevice.WriteByte(conf);
+
+            _i2cDevice.Read(_readBuffer);
+            ReadConfigByte(LastConfigByte);
+
+            if (_lastChannel != channel)
+            {
+                errorList?.Add($"Channel update failed from {_lastChannel} to {channel}");
+                ok = false;
+            }
+            if (Resolution != resolution)
+            {
+                errorList?.Add($"Resolution update failed from {Resolution} to {resolution}");
+                ok = false;
+            }
+            if (mode != Mode)
+            {
+                errorList?.Add($"Mode update failed from {Mode} to {mode}");
+                ok = false;
+            }
+            if (PGAGain != pgaGain)
+            {
+                errorList?.Add($"PGAGain update failed from {PGAGain} to {pgaGain}");
+                ok = false;
+            }
+
+            if (!ok)
+            {   // Only use console on error
+                errorList?.Add($"Sent config byte {conf:X}, received {LastConfigByte:X}");
+            }
+            return ok;
+        }
+
+        protected int WaitTime => (int)(1000.0 / Helpers.UpdateFrequency(Resolution));
+
+        public int ChannelCount { get; }
+
+        protected ConversionResult LastConversion
+        {
+            get
+            {
+                return _lastConversion;
+            }
+            set
+            {
+                _lastConversion = value;
+                OnConversion?.Invoke(this, _lastConversion);
+            }
+        }
+
+        public event EventHandler<ConversionResult> OnConversion;
+
+        protected void WriteConfig(byte configByte)
+        {
+            _i2cDevice.WriteByte(configByte);
+        }
+
+        private static int _asyncThreshold = 20;
+
+        /// <summary>
+        /// Sets the lower time limit in ms. If the current configuration reads data faster the synchronous API is used.
+        /// This can save some overhead.
+        /// </summary>
+        /// <remarks>
+        /// The default configuration is 20ms. This means that only in 16 bit resolution read waits asynchronously.
+        /// Setting it to 0 or lower disables the function.
+        /// </remarks>
+        /// <param name="thresh">Time limit in ms. Default: 20ms</param>
+        /// <autogeneratedoc />
+        public static void SetAsyncThreshold(int thresh) { _asyncThreshold = thresh; }
```

Comment:
why is this static? what if you have two devices connected but want different values for them?

Reply:
Because I imagined `_asyncThreshold ` as an application wide design decision where the developer can decide if their use case values non-blocking read but nondeterministic timing or blocking call with more strict timing.

I can't see real benefits of having different values for separate instances. Read speed itself is settable per instance by setting the resolution.

I can change it if you want, but I would find the above function confusing on an instance.